fiftyone/utils/eval/regression.py

```diff
@@ -163,12 +164,33 @@ def get_fields(self, samples, eval_key):
 
         return fields
 
+    def rename(self, samples, eval_key, new_eval_key):
```

Comment:
I was following along with how the `cleanup()` and `get_fields()` methods are implemented explicitly on each evaluation protocol.

The implementations for each protocol are currently similar, but there's not an inherent reason why they would always be the same, so I think some repeated code is okay in this case.

Reply:
See what you're saying but can't say I agree with the logic. I feel like this is why when we add something to OSS backend we will have multiple places we have to add it to. Which is probably fine if you're making the edit but anyone else might easily miss a spot.

Reply:
As it currently is, there are `XXXEvaluation` base classes for each type of evaluation protocol: classification, detection, regression, segmentation. 

The abstraction to remove duplicate code would be "this class of evaluation protocol supports either sample-level or frame-level fields. When processing sample-level fields, it populates a single field whose name matches the `eval_key`. When processing frame-level fields, it populates a frame-level and sample-level fields whose name matches `eval_key`." For that class of methods, we could abstract `get_fields()`, `rename()`, and `cleanup()` into a mixin and share the code.

It doesn't seem that useful to me though. Sometimes when there's a bunch of fancy subclass implementations it becomes less clear how to implement a new instance of the interface (eg a new label type to evaluate) because it seems like you have to follow how the base class does some things rather than just doing whatever you need for your case.